src/components/RefundButton.tsx

```diff
@@ -223,19 +227,38 @@ export const RefundBtc = (props: {
         if (!swap) {
             return undefined;
         }
+        try {
+            const transactionToRefund = await getLockupTransaction(
+                swap.id,
+                swap.type,
+            );
 
-        const transactionToRefund = await getLockupTransaction(
-            swap.id,
-            swap.type,
-        );
+            // show refund ETA for legacy swaps
+            if (swap.version !== OutputType.Taproot) {
+                setTimeoutEta(transactionToRefund.timeoutEta);
+                setTimeoutBlockheight(transactionToRefund.timeoutBlockHeight);
+            }
 
-        // show refund ETA for legacy swaps
-        if (swap.version !== OutputType.Taproot) {
-            setTimeoutEta(transactionToRefund.timeoutEta);
-            setTimeoutBlockheight(transactionToRefund.timeoutBlockHeight);
-        }
+            return transactionToRefund;
+        } catch {
+            const utxos = await fetchUTXOsWithFailover(
+                swap.assetSend,
+                swap.address,
+            );
 
-        return transactionToRefund;
+            if (utxos.length) {
```

Comment:
```suggestion
            if (utxos.length > 0) {
```

Not a fan of relying on 0 being falsy

Reply:
@michael1011 fixed this and the other cases where this is repeated